ceph/rados/serviceability_workflows.py

```diff
@@ -253,36 +317,15 @@ def remove_custom_host(self, host_node_name: str):
             # Sleeping for 2 seconds for removal to have started
             time.sleep(2)
             log.debug(f"Started drain operation on node : {rm_host.hostname}")
-
-            status_cmd = "ceph orch osd rm status -f json"
-            end_time = datetime.datetime.now() + datetime.timedelta(seconds=600)
-            flag = False
-            while end_time > datetime.datetime.now():
-                out, err = self.cephadm.shell([status_cmd])
-                try:
-                    drain_ops = json.loads(out)
-                    for entry in drain_ops:
-                        log.debug(
-                            f"Drain operations are going on host {rm_host.hostname} \nOperations: {entry}"
-                        )
-                except json.JSONDecodeError:
-                    log.info(f"Drain operations completed on host : {rm_host.hostname}")
-                    flag = True
-                    break
-                except Exception as error:
-                    log.error(f"Hit issue during drain operations: {error}")
-                    raise Exception(error)
-                log.debug("Sleeping for 10 seconds and checking again....")
-                time.sleep(10)
-
-            if not flag:
+            if wait_osd_operation_status(status_cmd):
+                log.info(
+                    f"Completed drain operation on the host. {rm_host.hostname}\n Removing host from the cluster"
+                )
+            else:
```

Comment:
just before removal, check if any services exist. line no 335.

Reply:
Included the check.